parquet/src/main/java/org/apache/iceberg/parquet/ParquetSchemaUtil.java

```diff
@@ -164,4 +164,48 @@ public Boolean primitive(PrimitiveType primitive) {
     }
   }
 
+  // Parquet LIST backwards-compatibility rules.
+  // https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#backward-compatibility-rules
+  public static boolean isOldListElementType(Type repeatedType, String parentName) {
```

Comment:
Does this need to be public? I don't see much value in Iceberg exposing it if we don't need to.

Reply:
It also seems like this is always used after extracting the repeated type. It might be easier just to pass in the list itself and return whether it is a 2-level list or not.